File: Examples/ObjcDemo/RixSDKPodDemo/RixEngineObjcDemo/Base/BaseUIViewController.m
#import "BaseUIViewController.h"

@implementation BaseUIViewController

- (UIButton *)createButtonWithTitle:(NSString *)title action:(SEL)action {
    UIColor *tintColor = [UIColor colorWithRed:76/255.0 green:190/255.0 blue:196/255.0 alpha:1.0];
    UIButton *button = [UIButton buttonWithType:UIButtonTypeSystem];
    [button setTitle:title forState:UIControlStateNormal];
    button.layer.cornerRadius = 8.0;
    button.layer.borderWidth = 1.5;
    button.layer.borderColor = tintColor.CGColor;
    button.translatesAutoresizingMaskIntoConstraints = NO;
    [button setTitleColor:tintColor forState:UIControlStateNormal];
    button.backgroundColor = [UIColor whiteColor];
    [button addTarget:self action:action forControlEvents:UIControlEventTouchUpInside];
    return button;
}

- (UILabel *)createLabel {
    UILabel *label = [[UILabel alloc] init];
    label.translatesAutoresizingMaskIntoConstraints = NO;
    label.textAlignment = NSTextAlignmentCenter;
    label.textColor = [UIColor blackColor];
    label.numberOfLines = 0;
    return label;
}

- (void)clearSubView:(UIView *)containerView {
    for (UIView *view in containerView.subviews) {
        [view removeFromSuperview];
    }
}

@end

File: Examples/SwiftDemo/RixSDKPodDemo/AlxAdsDemo/Base/BaseUIViewController.swift
//
//  BaseUIViewController.swift
//  AlxDemo
//
//  Created by liu weile on 2025/3/31.
//

import UIKit

public class BaseUIViewController: UIViewController {

   public func createButton(title:String,action: Selector) -> UIButton{
       let tintColor = UIColor(red: 76/255.0, green: 190/255.0, blue: 196/255.0, alpha: 1.000)
       let button = UIButton()
       button.setTitle(title, for: .normal)
       button.layer.cornerRadius = 8.0
       button.layer.borderWidth = 1.5
       button.layer.borderColor = tintColor.cgColor
       button.translatesAutoresizingMaskIntoConstraints=false
       button.setTitleColor(tintColor, for: .normal)
       button.backgroundColor = .white
       button.accessibilityNavigationStyle = .automatic
       button.addTarget(self, action: action, for: .touchUpInside)
       return button
   }

    
    public func createLabel() -> UILabel{
        let label=UILabel()
        label.translatesAutoresizingMaskIntoConstraints=false
        label.textAlignment = .center
        label.textColor = .black
        label.numberOfLines = .zero
        return label
    }
    
    public func clearSubView(_ constainerView:UIView){
        let views=constainerView.subviews
        for view in views {
            view.removeFromSuperview()
        }
    }
}
